packages/zosuss/src/SshBaseHandler.ts

```diff
@@ -58,22 +59,103 @@ export abstract class SshBaseHandler implements ICommandHandler {
     public async process(commandParameters: IHandlerParameters) {
         this.mHandlerParams = commandParameters;
 
-        const sshSessCfgOverride: IOverridePromptConnProps[] = [{
-            propertyName: "privateKey",
-            propertiesOverridden: ["password", "tokenType", "tokenValue", "cert", "certKey", "passphrase"]
-        }];
-        const sshSessCfg: ISshSession = SshSession.createSshSessCfgFromArgs(commandParameters.arguments);
-        const sshSessCfgWithCreds = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISshSession>(
-            sshSessCfg, commandParameters.arguments, {
-                parms: commandParameters,
-                propertyOverrides: sshSessCfgOverride,
-                supportedAuthTypes: [SessConstants.AUTH_TYPE_BASIC]
-            }
+        const sshSessCfgOverride: IOverridePromptConnProps[] = [
+            {
+                propertyName: "privateKey",
+                propertiesOverridden: [
+                    "password",
+                    "tokenType",
+                    "tokenValue",
+                    "cert",
+                    "certKey",
+                ],
+            },
+        ];
+        const sshSessCfg: ISshSession = SshSession.createSshSessCfgFromArgs(
+            commandParameters.arguments
         );
+        let sshSessCfgWithCreds =
+            await ConnectionPropsForSessCfg.addPropsOrPrompt<ISshSession>(
+                sshSessCfg,
+                commandParameters.arguments,
+                {
+                    parms: commandParameters,
+                    propertyOverrides: sshSessCfgOverride,
+                    supportedAuthTypes: [SessConstants.AUTH_TYPE_BASIC],
+                }
+            );
         this.mSession = new SshSession(sshSessCfgWithCreds);
 
         this.mArguments = commandParameters.arguments;
-        await this.processCmd(commandParameters);
+
+        try {
+            await this.processCmd(commandParameters);
+        } catch (e) {
+            this.console.log("Initial key passphrase authentication failed!" + "\n");
+            if (
+                e.message.includes("but no passphrase given") ||
+                e.message.includes("bad passphrase?")
+            ) {
+                const maxAttempts = 3;
+                let attempt = 0;
+                let success = false;
+                while (attempt < maxAttempts && !success) {
```

Comment:
Not suggesting any changes for this pull request, but I could see parts of this logic being valuable for "prompting w/ `x` attempts" in other areas of the CLI 😋

Maybe at some point and if the demand is there, this code could be generalized and pulled into a utility class/function so it can be leveraged for different features.